metanorm/normalizers/ftp.py

```diff
@@ -0,0 +1,146 @@
+"""Normalizer for the interpretation of file name convention"""
+
+import calendar
+import logging
+import re
+import dateutil.parser
+import pythesint as pti
+from dateutil.relativedelta import relativedelta
+from dateutil.tz import tzutc
+from django.contrib.gis.geos.geometry import GEOSGeometry
+from urllib.parse import urlparse
+from .base import BaseMetadataNormalizer
+from datetime import datetime
+
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.addHandler(logging.NullHandler())
+
+
+class URLMetadataNormalizer(BaseMetadataNormalizer):
+    """ Normalizer for hardcoding information based on URLS """
+
+    def looper(self, associated_dict, raw_attributes_url, pti_function_name):
+        """looper is a helper function. It loops through the keys of "associated_dict" for finding
+        the corresponding value based on 'url' and 'pti_function_name' method of pti"""
+        for url in associated_dict.keys():
+            if raw_attributes_url.startswith(url):
+                return pti.__getattribute__(pti_function_name)(associated_dict[url])
+
```

Comment:
1. Always name your functions with verb.
2. Avoid repetition
3. Simplify calls to functions
```suggestion
    def get_matching_value(self, associated_dict, raw_attributes, url_function):
        """ Loop through <associated_dict> and get matching value using appropriate function """
        if 'url ' in raw_attributes:
            for url in associated_dict.keys():
                if raw_attributes['url'].startswith(url):
                    return url_function(associated_dict[url])
```